### ISSUE TITLE
'router-link-active' not being applied on nested routes

### ISSUE THREAD

**Issue by experimental9320** [vuejs/vue-router-next#984]
### Version
4.0.8

### Reproduction link
[https://codesandbox.io/s/agitated-herschel-62rdy?file=/index.html](https://codesandbox.io/s/agitated-herschel-62rdy?file=/index.html)

### Steps to reproduce
Click the 'Foo > Bar' link.

### What is expected?
Both the 'Foo' and the 'Foo > Bar' link should both be colored red since the route is '/foo/bar', which is a nested route of '/foo', which is the path of the 'Foo' link.

### What is actually happening?
Only the 'Foo > Bar' link is colored red.

---
Basically, the 'router-link-active' class is behaving like the 'router-link-exact-active' class in that it only is applied with exact route matches. Instead, the 'router-link-active' class should be applied to the parents of nested routes.

**Comment by posva** [comment]
This is documented in the migration guide in the docs

**Comment by experimental9320** [comment]
Ah ok. Thank you for pointing that out. I apologize for not reading the documentation thoroughly enough.

For any future soul who runs into this problem and can't find the answer, here's the solution:

In reference to the "Unrelated but similar routes" in [this RFC](https://github.com/vuejs/rfcs/blob/master/active-rfcs/0028-router-active-link.md#unrelated-but-similiar-routes), you can set up your routes like this to have links of parent routes receive the `router-link-active` class:
```Typescript
// Vue 3
import { h } from 'vue'
import { RouterView } from 'vue-router'

import MovieList from '@/components/MovieList.vue'
import NewMovie from '@/components/NewMovie.vue'
import MovieSearch from '@/components/MovieSearch.vue'

const routes = [
  {
    path: '/movies',
    // we need this to render the children (see note below)
    component: { render: () => h(RouterView) },
    // for vue 2 use render: h => h('RouterView')
    children: [
      { path: '', component: MovieList },
      { path: 'new' , component: NewMovie},
      // different child
      { path: 'search', component: MovieSearch }
    ]
  }
]

```
